homeassistant/components/command_line/sensor.py

```diff
@@ -138,7 +138,7 @@ async def async_update(self) -> None:
             else:
                 _LOGGER.warning("Empty reply found when expecting JSON data")
 
-        elif self._value_template is not None:
+        if self._value_template is not None:
```

Comment:
I'm not sure what you're asking for here, unless I'm mistaken value_template should still work with non-JSON output as it did previously.

The old code, before PR 90268, had this:

```Python
if self._json_attributes:
    ... snipped attributes code ...

if value is None:
    value = STATE_UNKNOWN
elif self._value_template is not None:
```
That PR removed the `if value is None` part, so None would be used in place of STATE_UNKNOWN but left the following `elif`, so it became part of the `if self._json_attributes:` block and would only apply if json_attributes was not set, rather than making it `if` so that value_template is handled regardless of whether json_attributes is set or not.

Reply:
IMO You create a new issue with the else at L148-L149.
We should not set the native value if `json_attributes` is set and a dict is expected, and `value_template` is not set.

Reply:
It looks like that change would result in the state being unknown if the JSON attributes don't include **state**, even if there is a value_template defined.

Would doing this be OK as a compromise?

```Python
self._attr_native_value = (
      json_dict[CONF_STATE] if CONF_STATE in json_dict else value
)
```

So that it would default to the value from rendering value_template.

Reply:
> Would doing this be OK as a compromise?

If `json_attributes` is set value is a assumed to be a `dict`. We do not want to write that as a state too. When a `value_template`  is set, then the state could be rendered to a simple value (only when json_attributes is not set). My suggesting is to make the `value_template` useful for rendering a non json output to a json output so that `json_attributes` can be used to in that case.

Reply:
I'm not sure that it makes sense to use value_template to render a non-JSON output to JSON, as the main function of it is to render to the output to a simple value. For example, one of my sensors calls a script which returns multiple values in an array in a JSON object (e.g. `{"names":["Test1","Test2"]}`.
the value_template is used so the latest value is the state of the sensor, and the json_attributes has an attribute returning all the versions.

Making this change would require users to change the scripts so that the JSON they return would include a "state" attribute, if they are using json_attributes and don't want the value of their sensor to be "unknown".

Reply:
I've implemented the original suggestion now, so if there is no value_template, but json_attributes are used the state will be None (to avoid the state being populated with the raw JSON data). I think for now that is the simplest thing to do and it still resolves the original issue.